### PR TITLE
set maximum values for stormcraft heat to energy

### PR DESCRIPTION
**Pull request by bafolts**
For #1794 . Sets the maximum value to a maximum of 8 or 4. Also added error checking for over paying.

### CODE REVIEW

**Review by kberg — COMMENTED** [review]
I would like to say approved, but I'm not quite sure I follow the logic changes here. A test would help a lot.

File: src/Player.ts [hunk 1]
```diff
@@ -1726,19 +1726,19 @@ export class Player implements ILoadable<SerializedPlayer, Player>{
...
-      let heatAmount: number;
-      let floaterAmount: number;
       if (this.isCorporation(CardName.STORMCRAFT_INCORPORATED) && this.getResourcesOnCorporation() > 0 ) {
+        let heatAmount: number;
+        let floaterAmount: number;
         const raiseTempOptions = new AndOptions (
           () => {
-            if (heatAmount + (floaterAmount * 2) < 8) {
+            const total = heatAmount + (floaterAmount * 2);
+            if (total < 8) {
```
Comment (nwai90) [inline src/Player.ts:1738]: 👍 We could also use `constants.HEAT_FOR_TEMPERATURE` here and in all places in this PR that have the hardcoded value of 8

**Review by nwai90 — APPROVED** [review]

File: src/Player.ts (hunk 1 above)
Comment (bafolts) [reply src/Player.ts:1738]: OK. I will also add test. The logic change here is that one can only spend 8 heat or 4 floaters to raise temperature. The maximum amount they can spend is then 8 or 4, regardless of how much heat or floaters on card they have.

Comment (kberg) [reply src/Player.ts:1738]: BTW, sorry -- I should have said, I'm not dying for a test here, but clarity. I rescind any request for a test, implied or explicit.

Comment (bafolts) [reply src/Player.ts]: Ask and ye shall receive. I moved the conversion function onto the card to ease writing tests.

**Pull request merged**